Sources/AblyAssetTrackingInternal/AblyWrapper/DefaultAbly.swift

```diff
@@ -129,38 +132,123 @@ public class DefaultAbly: AblyCommon {
             completion(.success(false))
             return
         }
-        
+
+        do {
+            try performChannelOperationWithRetry(channel: channelToRemove, operation: { channel in
+                guard let presenceData = presenceData else {
+                    return
+                }
+
+                try disconnectChannel(channel: channel, presenceData: presenceData)
+            })
+            
+            channels.removeValue(forKey: trackableId)
+            completion(.success(true))
+        }catch AblyError.connectionError(let errorInfo){
+            completion(.failure(errorInfo.toErrorInformation()))
+        }catch{
+            completion(.failure(ErrorInformation(error: AblyError.connectionError(errorInfo: ARTErrorInfo.create(withCode: 100_000, message: "Unknown error while disconnecting")))))
+        }
+    }
+    
+    private func disconnectChannel(channel:AblySDKRealtimeChannel, presenceData: PresenceData) throws{
+        leavePresence(channel: channel, presenceData: presenceData) { leaveError in
+            if let leaveError = leaveError {
```

Comment:
What's the purpose of throwing this error, given that `leavePresence` ignores it?

Reply:
Wouldn't this error be propagated to the user 🤔

Reply:
As discussed offline let's take a look at this code together

Reply:
I changed the general flow to use completion handler based error propagation rather than using direct throws which was making the whole change messy : 248ff80d6b48c0c988fa637c81c01c9ded685831